projects/testing-library/src/lib/models.ts

```diff
@@ -231,7 +237,7 @@ export interface RenderComponentOptions<ComponentType, Q extends Queries = typeo
    *  ]
    * })
    */
-  componentProviders?: any[];
+  componentProviders?: Array<ProviderInterface | TypeProvider | EnvironmentProviders>;
```

Comment:
I think this type definition is not correct. Look here:

https://github.com/testing-library/angular-testing-library/blob/main/projects/testing-library/src/lib/testing-library.ts#L104-L109

`componentProviders` are passed into `TestBed.overrideProvider(provide, provider)`. So each item of that MUST have `provide` property and it is not a TypeProvider. 

I guess this array should be typed just as `ValueProvider | ClassProvider | ExistingProvider | FactoryProvider`.

Reply:
Thanks for the feedback. I updated the PR. Additionally, I added some of the more types added in https://github.com/angular/angular/pull/51766/files